tests/proc/lif/test_models.py

```diff
@@ -0,0 +1,350 @@
+        super().__init__(**kwargs)
+        shape = kwargs.get("shape", (1,))
+        self.shape = shape
+        self.s_in = InPort(shape=(shape[1],))
+        self.spk_data = Var(shape=shape, init=0)  # This Var expands with time
+
+
+@implements(proc=VecSendProcess, protocol=LoihiProtocol)
+@requires(CPU)
+# need the following tag to discover the ProcessModel using LifRunConfig
+@tag('floating_pt')
+class PyVecSendModel1(PyLoihiProcessModel):
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, float)
+    vec_to_send: np.ndarray = LavaPyType(np.ndarray, float)
+    send_at_times: np.ndarray = LavaPyType(np.ndarray, bool, precision=1)
+
+    def run_spk(self):
+        """
+        Send `spikes_to_send` if current time-step requires it
+        """
+        if self.send_at_times[self.current_ts-1]:
+            self.s_out.send(self.vec_to_send)
+        else:
+            self.s_out.send(np.zeros_like(self.vec_to_send))
+
+
+@implements(proc=VecSendProcess, protocol=LoihiProtocol)
+@requires(CPU)
+# need the following tag to discover the ProcessModel using LifRunConfig
+@tag('fixed_pt')
+class PyVecSendModel2(PyLoihiProcessModel):
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, np.int16, precision=16)
+    vec_to_send: np.ndarray = LavaPyType(np.ndarray, np.int16, precision=16)
+    send_at_times: np.ndarray = LavaPyType(np.ndarray, bool, precision=1)
+
+    def run_spk(self):
+        """
+        Send `spikes_to_send` if current time-step requires it
+        """
+        if self.send_at_times[self.current_ts-1]:
+            self.s_out.send(self.vec_to_send)
+        else:
+            self.s_out.send(np.zeros_like(self.vec_to_send))
+
+
+@implements(proc=VecRecvProcess, protocol=LoihiProtocol)
+@requires(CPU)
+# need the following tag to discover the ProcessModel using LifRunConfig
+@tag('floating_pt')
+class PySpkRecvModel1(PyLoihiProcessModel):
+    s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
+    spk_data: np.ndarray = LavaPyType(np.ndarray, float)
+
+    def run_spk(self):
+        """Receive spikes and store in an internal variable"""
+        spk_in = self.s_in.recv()
+        self.spk_data[self.current_ts-1, :] = spk_in
+
+
+@implements(proc=VecRecvProcess, protocol=LoihiProtocol)
+@requires(CPU)
+# need the following tag to discover the ProcessModel using LifRunConfig
+@tag('fixed_pt')
+class PySpkRecvModel2(PyLoihiProcessModel):
+    s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
+    spk_data: np.ndarray = LavaPyType(np.ndarray, int, precision=1)
+
+    def run_spk(self):
+        """Receive spikes and store in an internal variable"""
+        spk_in = self.s_in.recv()
+        self.spk_data[self.current_ts-1, :] = spk_in
+
+
+class TestLIFProcessModels(unittest.TestCase):
+    """Tests for all ProcessModels of LIF"""
+    def test_float_pm_no_decay(self):
+        """
+        Set up bias = 1 * 2**1 = 2. and threshold = 4. Without any current
```

Comment:
Often also useful to document unit test code inline to show intent so it's easier to understand in half a year.

Reply:
Done.